sw/device/tests/aes_entropy_test.c

```diff
@@ -32,20 +28,33 @@ static const uint8_t kKey[] = {
     0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f,
 };
 
-static const uint32_t kCipherTextGold[] = {
-    0x493350e6,
-    0x4a0a5568,
-    0x9d4d4aa1,
-    0xd256e2e0,
-};
-
 static const uint32_t kIv[] = {
     0x01020304,
     0x0c0d0e0f,
     0x1718191a,
     0x1c1d1e1f,
 };
 
+/**
+ * The following ciphertext are computed using the python library `PyCryptodome`
+ * using the above plainText, key and iv.
+ *
+ * ```Python
+ *  from Crypto.Cipher import AES
+ *  key = bytes.fromhex(("000102030405060708090a0b0c0d0e0f1"
+ *                      "01112131415161718191a1b1c1d1e1f"))
+ *  iv = bytes.fromhex("040302010f0e0d0c1a1918171f1e1d1c")
+ *  msg = bytes.fromhex("00112233445566778899aabbccddeeff")
+ *  print(AES.new(key, AES.MODE_CBC, iv).encrypt(msg).hex())
+ * ```
+ */
```

Comment:
Not a big deal (i.e. we can update this later, with an issue filed to track this), but it is preferred to pick NIST vectors that are known to be correct by an industry standard, rather than rely on a third party library. See 
https://github.com/lowRISC/opentitan/blob/ee49544c240fd58a9568877c46e6dfbb5260334d/sw/device/lib/testing/hmac_testutils.h#L46

for the HMAC example.

Reply:
Ohh wait - I just realized it _was_ already a NIST vector but the cipher text did not match. I feel it's worth understanding why it did not match.

Reply:
The original NIST vector was from a ECB examble, but this test must be CBC because of the `iv`.
I could't quickly find a NIST reference for CBC so I decided to use the PyCryptodome.

Reply:
See https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38a.pdf and https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Standards-and-Guidelines/documents/examples/AES_ModesA_All.pdf for AES NIST vectors (all modes incl. CBC), we already have them in the source tree (twice):
- https://github.com/lowRISC/opentitan/blob/ee49544c240fd58a9568877c46e6dfbb5260334d/hw/ip/aes/model/aes_modes.h
- https://github.com/lowRISC/opentitan/blob/ee49544c240fd58a9568877c46e6dfbb5260334d/hw/ip/aes/dv/test_vectors/nist_vectors_pkg.sv

Reply:
Thank you very much @vogelpi. I've changed the test to use the vectors directly from the `hw/ip/aes/model/aes_modes.h` file.